src/Region/Region.cc

```diff
@@ -638,36 +680,47 @@ std::vector<std::vector<CARTA::Point>> Region::GetApproximatePolygonPoints(int n
     CARTA::Point first_point(region_points[0]);
     region_points.push_back(first_point);
 
+    // Total length (pixels) of reference region segments
     double total_length = GetTotalSegmentLength(region_points);
-    double target_segment_length = total_length / num_vertices;
-
-    // Divide each region polygon segment into target number of segments with target length
-    for (size_t i = 1; i < region_points.size(); ++i) {
-        // Handle segment from point[i-1] to point[i]
-        std::vector<CARTA::Point> segment_points;
-
-        auto delta_x = region_points[i].x() - region_points[i - 1].x();
-        auto delta_y = region_points[i].y() - region_points[i - 1].y();
-        auto segment_length = sqrt((delta_x * delta_x) + (delta_y * delta_y));
-        auto dir_x = delta_x / segment_length;
-        auto dir_y = delta_y / segment_length;
-        auto target_nsegment = round(segment_length / target_segment_length);
-        auto target_length = segment_length / target_nsegment;
-
-        auto first_segment_point(region_points[i - 1]);
-        segment_points.push_back(first_segment_point);
-
-        auto first_x(first_segment_point.x());
-        auto first_y(first_segment_point.y());
-
-        for (size_t j = 1; j < target_nsegment; ++j) {
-            auto length_from_first = j * target_length;
-            auto x_offset = dir_x * length_from_first;
-            auto y_offset = dir_y * length_from_first;
-            segment_points.push_back(Message::Point(first_x + x_offset, first_y + y_offset));
-        }
+    double min_length = 0.1;
```

Comment:
@kswang1029 here is where the minimum segment length between polygon points is set to 0.1 pixels.

Reply:
I commented out this length and the maximum error was still 10.25%.  Will look into implementation details particularly at the box corners.